back/engines/commercial/analysis/app/lib/analysis/auto_tagging_method/base.rb

```diff
@@ -160,6 +160,21 @@ def deduct_locale(input)
         AppConfiguration.instance.settings('core', 'locales').first
     end
 
+    def classification_fields
+      fields = analysis.associated_custom_fields
+      if fields.map(&:code).include?('topic_ids') || analysis.participation_method != 'ideation'
+        fields
+      else
+        custom_form = analysis.project.custom_form || CustomForm.new(participation_context: analysis.project)
+        project_fields = IdeaCustomFieldsService.new(custom_form).submittable_fields
+        fields + [project_fields.find { |field| field.code == 'topic_ids' }].compact
+      end
+    end
+
+    def input_to_text_classify
```

Comment:
Why is this `InputToText` instance computed in a memoized method and the other one is computed in the initializer?